sdk/clientcore/core/src/main/java/io/clientcore/core/http/models/RequestOptions.java

```diff
@@ -113,42 +113,57 @@
  * <!-- end io.clientcore.core.http.rest.requestoptions.postrequest -->
  */
 @Metadata(properties = MetadataProperties.FLUENT)
-public final class RequestOptions {
-    // RequestOptions is a highly used, short-lived class, use a static logger.
+public class RequestOptions {
```

Comment:
Something I've been meaning to ask, should we call this `HttpRequestOptions` as this is HTTP specific and we've prefixed `Http` for everything else HTTP specific?

@JonathanGiles @srnagar

Reply:
Since `RequestOptions` will be used in every "WithResponse" method, we wanted to keep the type name short and also match `Response<T>`. Both `Request*` and `Response` don't have `Http` prefix and are types that users directly see on the client methods. 

That's the reason for not having the Http prefix. However, if we have to add it, we should add to both.